### PR TITLE
Refactor metadata

### PR DESCRIPTION
**Pull request by delarocamcas**
changes:
server.py: changed read method to return same data structure as base read exif tags function, update write method to only allow configured exif tags. 
test/test_server.py: update by read and write test for new metadata formats, created test for incorrect metadata request.

### CODE REVIEW

File: Dockerfile
Comment (delarocamcas) [inline Dockerfile:1]: dockerfile step to install submodule requirements

File: server.py [hunk 1]
```diff
@@ -553,13 +543,22 @@ def updateexifdata():
             abort(400)
```
Comment (delarocamcas) [inline server.py:544]: 422 error for incorrect tags, let me know if  a different error code is preffered

File: tests/test_server.py [hunk 2]
```diff
@@ -40,10 +40,9 @@ def get_file_md5(filename):
...
 
 EXIF_DICT = {
-      "33432": "\u00A9 california academy of sciences",
-        "315": "Claude Monet",
-        "33437": "3/2"
-
+       "IPTC:CopyrightNotice": "\u00A9 california academy of sciences",
+       "XMP:Creator": "Claude Monet",
+       "EXIF:CreateDate": "2023:02:02 13:48:41"
 }
```
Comment (delarocamcas) [inline tests/test_server.py:47]: updated test with data format to fit returns of updated exiftools

File: tests/test_server.py [hunk 3]
```diff
@@ -240,18 +240,45 @@ def test_update_exifdata():
...
+    assert exif_data['EXIF:CreateDate'] == '2023:02:02 13:48:41'
+
+    assert exif_data['XMP:Creator'] == 'Claude Monet'
+
+    assert exif_data['IPTC:CopyrightNotice'] == "\u00A9 california academy of sciences"
+
+    r = delete_attach_loc()
+
+    assert r.status_code == 200
+
```
Comment (delarocamcas) [inline tests/test_server.py:253]: unittests for bad write exif request

File: server.py.orig
Comment (foozleface) [inline server.py.orig:1]: we should remove this. probably one of mine, I use the .orig suffix when testing.

File: server.py [hunk 4]
```diff
@@ -553,13 +543,22 @@ def updateexifdata():
...
-            mt.write_exif_tags(exif_dict=exif_data)
+            md = MetadataTools(path=rel_path)
+            try:
+                md.write_exif_tags(exif_dict=exif_data)
+            except:
+                response.content_type = 'text/plain; charset=utf-8'
+                response.status = 422
+                response.body = f"422 - forbidden. " \
+                                f"Changes to following metadata tags not permitted: '{request.query.token}'"
+                log(response.body)
```
Comment (foozleface) [inline server.py:554]: This could be the response for an invalid tag or a tag that's valid that we don't have listed, so it's a little misleading. Something like "Tag not supported" might be more illustrative.

File: tests/test_server.py [hunk 5]
```diff
@@ -40,10 +40,9 @@ def get_file_md5(filename):
 TEST_DATE = dto
 
 EXIF_DICT = {
-      "33432": "\u00A9 california academy of sciences",
-        "315": "Claude Monet",
-        "33437": "3/2"
-
+       "IPTC:CopyrightNotice": "\u00A9 california academy of sciences",
```
Comment (foozleface) [inline tests/test_server.py:43]: I'd use the consts here - IPTC_COPYRIGHT_NOTICE, for example. Or a mix to test both?

File: tests/test_server.py [hunk 6]
```diff
@@ -215,13 +214,14 @@ def test_update_exifdata():
...
 
     exif_data = get_exif_data()
+
     # checking field not present before function execution
 
-    assert 'Copyright' not in exif_data[0]['Fields']
+    assert exif_data['EXIF:CreateDate'] == '2022:02:02 13:48:41'
 
-    assert 'Artist' not in exif_data[0]['Fields']
+    assert 'XMP:Creator' not in exif_data.keys()
```
Comment (foozleface) [inline tests/test_server.py:222]: same with all of these - XMP_CREATOR, for example.

File: tiny.py
Comment (foozleface) [inline tiny.py:1]: I think this is also mine - testing an edge case. Nuke it for me?

**Review by foozleface — COMMENTED** [review]
ok to land after indicated changes, no second review required.

File: server.py (hunk 4 above)
Comment (delarocamcas) [reply server.py:554]: ok i'll clarify the response message, is the 422 code still fine?
